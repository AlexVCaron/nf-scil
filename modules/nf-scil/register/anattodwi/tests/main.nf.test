nextflow_process {

    name "Test Process REGISTER_ANATTODWI"
    script "../main.nf"
    process "REGISTER_ANATTODWI"

    tag "modules"
    tag "modules_nfcore"
    tag "register"
    tag "register/anattodwi"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    config "./nextflow.config"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "processing.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("register - anattodwi") {
        when {
            process {
             """
             input[0] = LOAD_DATA.out.test_data_directory
                     .map{ test_data_directory -> [
                     [ id:'test', single_end:false ],
                     file("\${test_data_directory}/mni_masked_2x2x2.nii.gz"),
                     file("\${test_data_directory}/b0_mean.nii.gz"),
                     file("\${test_data_directory}/fa.nii.gz")
             ]}
             """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
