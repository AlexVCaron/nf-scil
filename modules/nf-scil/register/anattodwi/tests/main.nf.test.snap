{
    "register - anattodwi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output0GenericAffine.mat:md5,043decee0d131e79a802e3dcc61c22f0",
                        "test__output1InverseWarp.nii.gz:md5,62d6287e60dd36ffd477e71061f23077"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output1Warp.nii.gz:md5,f896e4fef21f26c84bf834afd7fb5d55",
                        "test__output0GenericAffine.mat:md5,043decee0d131e79a802e3dcc61c22f0"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_warped.nii.gz:md5,ff651c76f4ea493de0a06a202e9f9fb2"
                    ]
                ],
                "3": [
                    "versions.yml:md5,6c0ee74526552d351c4f525ae8737a84"
                ],
                "t1_warped": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_warped.nii.gz:md5,ff651c76f4ea493de0a06a202e9f9fb2"
                    ]
                ],
                "transfo_image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output1Warp.nii.gz:md5,f896e4fef21f26c84bf834afd7fb5d55",
                        "test__output0GenericAffine.mat:md5,043decee0d131e79a802e3dcc61c22f0"
                    ]
                ],
                "transfo_trk": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__output0GenericAffine.mat:md5,043decee0d131e79a802e3dcc61c22f0",
                        "test__output1InverseWarp.nii.gz:md5,62d6287e60dd36ffd477e71061f23077"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,6c0ee74526552d351c4f525ae8737a84"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-01T13:53:14.585418"
    }
}